src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -252,4 +253,80 @@ public void doUpdateNode(AbstractNode node) {
             return node;
         });
     }
+
+    public UUID getStudyRootNodeUuid(UUID studyId) {
+        List<NodeEntity> nodes = nodesRepository.findAllByStudyId(studyId);
+        if (nodes.isEmpty()) {
+            throw new StudyException(ELEMENT_NOT_FOUND);
+        }
+        return nodes.stream().filter(n -> n.getType().equals(NodeType.ROOT)).findFirst().orElseThrow().getIdNode();
+    }
+
+    @Transactional
+    public Optional<String> doGetVariantId(UUID id, boolean generateId) {
+        AbstractNode node = nodesRepository.findById(id).map(n -> repositories.get(n.getType()).getNode(id)).orElseThrow(() -> new StudyException(ELEMENT_NOT_FOUND));
+        if (node.getType() == NodeType.ROOT) {
+            return Optional.of("");  // we will use the network initial variant
+        } else if (node.getType() == NodeType.NETWORK_MODIFICATION) {
+            NetworkModificationNode networkModificationNode = (NetworkModificationNode) node;
+            if (networkModificationNode.getVariantId() == null && generateId) {
+                networkModificationNode.setVariantId(UUID.randomUUID().toString());  // variant id generated with UUID format ????
+                repositories.get(networkModificationNode.getType()).updateNode(networkModificationNode);
+            }
+            return Optional.of(networkModificationNode.getVariantId());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    public Mono<String> getVariantId(UUID id) {
+        return Mono.fromCallable(() -> self.doGetVariantId(id, true).orElse(null))
+            .switchIfEmpty(Mono.error(new StudyException(ELEMENT_NOT_FOUND)));
+    }
+
+    @Transactional
+    public Optional<UUID> doGetModificationGroupUuid(UUID id, boolean generateId) {
```

Comment:
Is the boolean parameter `generateId` necessary since it is set to `true` in all calls to `doGetModificationGroupUuid` ?

Reply:
doGetModificationGroupUuid can be further called with false as second parameter, when we will only
want to get the existing modification group uuid (when deleting nodes for example)

Reply:
Ok